WordPress/Classes/Models/Blog+Post.swift

```diff
@@ -28,3 +30,82 @@ public extension Blog {
         return (try? context.fetch(request))?.first
     }
 }
+
+// MARK: - Create posts
+
+extension Blog {
+
+    /// Create a post in the blog.
+    @objc
+    func createPost() -> Post {
+        guard let context = managedObjectContext else {
+            fatalError("The `Blog` instance is not associated with a `NSManagedObjectContext`")
+        }
```

Comment:
Would you consider using `throws` here?

My rule of thumb for using `fatalError` is whenever we run into inconsistent state because of a developer error. Classic example `URL(string: "invalid URL string")!`.

I don't know whether this situation counts as a developer error. I mean, one could argue that any error is at its root a developer error 😅 But I wonder if there are runtime instances in which this could occur because of race conditions or other bugs.

Anyway, this is just a question, not a request for change. Even if we were to adopt `throws` (or `Result<Post, Error>`) here, we'd be left with the question of how to handle the error at the call site for this method. 	Ideally, we'd bubble up the error till a point in which the UI can handle it, or where we can make it part of existing error handling,  but that seems way out of scope for the kind of low level infrastructure work done here.

The fact that this method is `@objc` also complicates the matter because it might restrict the kind of error propagation we might do.

Reply:
I think this case, where a `Blog` instance doesn't reference to any context, _is_ a programming error. Like, `let blog = Blog()` is incorrect, we should always create a new instance by `let blog = Blog(context: ...)` or query the blog from a context object.